File: gcd.erl
-module(gcd).
-export([gcd/2]).

gcd(A, B) when A == B ->
    A;
gcd(A, B) when A > B ->
    gcd(A-B, B);
gcd(A, B) when A < B ->
    gcd(B, B-A).

File: gcd.ex
defmodule Gcd do

  def gcd(a, b) when a == b do
    a
  end

  def gcd(a, b) when a > b do
    gcd(a-b, b)
  end

  def gcd(a, b) when a < b do
    gcd(b, b-a)
  end

end
